### PR TITLE
Align runtime registration for opencl and vulkan

### PR DESCRIPTION
**Pull request by abialas1**

### CODE REVIEW

File: pmlc/rt/vulkan/vulkan_runtime.cc [hunk 1]
```diff
@@ -54,6 +59,11 @@ class VulkanRuntime final : public pmlc::rt::Runtime {
   std::vector<std::shared_ptr<VulkanDevice>> devices;
 };
 
-pmlc::rt::RuntimeRegistration<VulkanRuntime> reg{"vulkan"};
+extern void registerSymbols();
+
+void registerRuntime() {
+  registerSymbols();
+  registerRuntime("vulkan", std::make_shared<VulkanRuntime>());
```
Comment (flaub) [inline pmlc/rt/vulkan/vulkan_runtime.cc:66]: We could just put a try/catch around this line I think.

Comment (flaub) [reply pmlc/rt/vulkan/vulkan_runtime.cc:66]: Actually, perhaps we should do it like:
```
try {
  registerRuntime("vulkan", std::make_shared<VulkanRuntime>());
  registerSymbols();
} catch (std::exception& ex) {
  IVLOG(2, "Failed to register 'vulkan' runtime: " << ex.what());
}
```

This way we only call `registerSymbols` if the runtime can be constructed.

Comment (abialas1) [reply pmlc/rt/vulkan/vulkan_runtime.cc:66]: Yes could be, however I wanted to be consistent with the OpenCL runtime as is. If you think it is important I can move it to this place.

File: pmlc/rt/opencl/opencl_runtime.cc [hunk 2]
```diff
@@ -29,6 +30,13 @@ class OpenCLRuntime final : public pmlc::rt::Runtime {
...
 };
 
-pmlc::rt::RuntimeRegistration<OpenCLRuntime> reg{"opencl"};
+extern void registerSymbols();
+extern void registerMemrefCasts();
+
+void registerRuntime() {
+  registerSymbols();
+  registerMemrefCasts();
+  registerRuntime("opencl", std::make_shared<OpenCLRuntime>());
```
Comment (flaub) [inline pmlc/rt/opencl/opencl_runtime.cc:39]: Might as well try/catch here too.

Comment (abialas1) [reply pmlc/rt/opencl/opencl_runtime.cc:39]: Ok let me change both vulkan and ocl to what you proposed

**Review by flaub — APPROVED** [review]

**Pull request merged**